### PR TITLE
feat: queue.purge and queue.approxsize

### PR DESCRIPTION
**Pull request by revitalbarletz**
This change introduces two new methods: 1. Queue.Purge which removes all messages from the queue. 2. Queue.ApproxSize which returns the approximate size of the queue.

*By submitting this pull request, I confirm that my contribution is made under the terms of the 
[Monada Contribution License](https://docs.winglang.io/terms-and-policies/contribution-license.html)*.

### CODE REVIEW

File: libs/wingsdk/src/target-sim/queue.inflight.ts [hunk 1]
```diff
@@ -40,6 +40,27 @@ export class Queue implements IQueueClient, ISimulatorResourceInstance {
...
   }
 
+  public async purge(): Promise<void> {
+    return this.context.withTrace({
+      message: `Purge ().`,
+      activity: async () => {
+        // TODO: which one is better?
+        this.messages.length = 0;
+        //this.messages.slice(0);
+        //this.messages = new Array<string>();
```
Comment (Chriscbr) [inline libs/wingsdk/src/target-sim/queue.inflight.ts:50]: Let's remove these extra comments

**Review by Chriscbr — COMMENTED** [review]
Looks great @revitalbarletz ! Just one small comment.

BTW - we are using a linter that expects all pull request titles to be all-lowercase. So we should rename `queue.approxSize` in the title to `queue.approx_size`

File: libs/wingsdk/src/target-sim/queue.inflight.ts (hunk 1 above)
Comment (revitalbarletz) [inline libs/wingsdk/src/target-sim/queue.inflight.ts:50]:
```suggestion
```

File: libs/wingsdk/src/target-sim/queue.inflight.ts [hunk 2]
```diff
@@ -40,6 +40,27 @@ export class Queue implements IQueueClient, ISimulatorResourceInstance {
     });
   }
 
+  public async purge(): Promise<void> {
+    return this.context.withTrace({
+      message: `Purge ().`,
+      activity: async () => {
+        // TODO: which one is better?
```
Comment (revitalbarletz) [inline libs/wingsdk/src/target-sim/queue.inflight.ts:47]:
```suggestion
```

**Review by Chriscbr — APPROVED** [review]

File: libs/wingsdk/src/target-sim/queue.inflight.ts (hunk 1 above)
Comment (revitalbarletz) [reply libs/wingsdk/src/target-sim/queue.inflight.ts]: Done.

File: libs/wingsdk/src/target-sim/queue.inflight.ts [hunk 3]
```diff
@@ -40,6 +40,24 @@ export class Queue implements IQueueClient, ISimulatorResourceInstance {
     });
   }
 
+  public async purge(): Promise<void> {
+    return this.context.withTrace({
+      message: `Purge ().`,
+      activity: async () => {
+        this.messages.length = 0;
```
Comment (eladb) [inline libs/wingsdk/src/target-sim/queue.inflight.ts:47]: Wow TIL

**Review by eladb — COMMENTED** [review]
Nice!

Revital awesome work!

**Pull request merged**